posts/2024-02-08-interviews-anirban_chetia/index.qmd

```diff
@@ -0,0 +1,161 @@
+
+
+## Areas of Improvement
+
+### Technical
+
+- An R core member dealing gingerly pointed out that after several years of abundant reports of installation problems on MacOS, `data.table` still shows as being unable to detect OpenMP support and use multiple threads on the platform (while the same is not prevalent on Windows or Linux), even after including OpenMP run-time in CRAN R releases specifically for `data.table`. I noticed the same issue too (and it has been there for a while as it appears, as I first [encountered](https://github.com/Anirban166/Mac-Issues/issues/2) it more than two years ago) on OS X being prevalent till date.
+
+- People miss `fread` being able to read fixed-width files, although there is `iotools` now.
+
+- Additional functionality for working with spatial geometries and mixed-model packages (such as `glmmTMB` and `lme4`).
+
+- Some people found using the Walrus operator to be weird, especially given that not every `data.table` operation requires that. They don't like keeping in mind the names of columns (they tend to move towards the set function for explicitly assigning) as well. The in-place assignment using `:=` is acceptable as they feel, but they would ideally want to be able to do `DT$x <- y` meaning `DT[, x := y]`.
+
+- `dataset[get("categoricalColumn")]` could be optimized further.
+
+- Efficient visualization methods can be introduced, if applicable or intuitive.
+
+- A few people find the syntax of `dcast` and `melt` to be confusing, and often end up making mistakes since those reshaping operations complement each other (long to wide and vice versa respectively) or are the reverse. Probably not best to change this given it would break things and is just something to be learned over time, but more examples might help.
+
+- More often a mild inconvenience than a common source of error, but to a few, the masking of other packages is something they do not like.
+
+### Non-technical
+
+- Documentation tends to be lacking, i.e. not enough well-documented `data.table` resources or online materials (blogs/articles, videos) exist. Even experienced developers feel that it isn't entirely straightforward to find out how to do more complex things, so more extensive documentation and examples would be great, if not a necessity. Making existing documentation more lucid and navigable, while less aggressive for certain phrases out of old-school culture, is another point mentioned by a few. Things that read as friendly and expressive while maintaining details are a go-to.
```

Comment:
can you please clarify " while less aggressive for certain phrases out of old-school culture" ?

Reply:
This is something that Nic said, and as per my notes she meant about certain statements in the FAQ (didn't ask her further)

Reply:
I'll exclude that part since I do not have a more specific example from her to clarify, and since I already mentioned a similar point in another section (under 'Barriers in contributing')